src/client.py

```diff
@@ -0,0 +1,362 @@
+        See https://mermaid-api.readthedocs.io/en/latest/nonproject.html for more information on non-project resources.
+        :param info: health, version, profiles, projecttags, sites, managements, me.
+        :type info: str
+        :return: non-project resource data.
+        :rtype: dict, str, None
+        """
+        # TODO: Handle pagination, filters, PUT, HEAD OPTIONS requests
+        # Authentication required for access to 'me' API endpoint.
+        if info == 'me' and not self.authenticated:
+            return None
+
+        if info in self.npr_endpoints:
+            return self.api_root(info)
+        return None
+
+    def get_choices(self):
+        """
+        Gets a list of objects, each one of which has a name item (e.g., countries) and a data item that is a list
+        of available choice objects.
+        :return: list of choice objects.
+        :rtype: list
+        """
+        # TODO: Filter: /updates/?timestamp="DATE"
+        return self.api_root('choices')
+
+    def get_attribute(self, attr):
+        """
+        Gets attributes which are “things that can be observed” – coral and other taxa as well as non-organic
+        benthic substrates for benthic transects and bleaching surveys, fish species/genera/families as well
+        as arbitrary fish groupings for fish belt transects, and so on.
+        :param attr: fishsizes, benthicattributes, fishfamilies, fishgenera, fishspecies, fishgroupings.
+        :type attr: str
+        :return: attribute data.
+        :rtype: dict, None
+        """
+        if attr not in self.attrs_endpoints:
+            return None
+        return self.api_root(attr)
+
+    def get_projects(self, showall=False):
+        """
+        Gets the projects subdirectory. Without query parameters, returns a list of projects of which
+        the user is a member. The showall query parameter may be used to return projects unfiltered by the user’s
+        membership. showall is important when the user is unauthenticated.
+        :param showall: (optional) Returns all projects unfiltered by the user’s membership.
+        :type showall: bool
+        :return: projects.
+        :rtype: dict, None
+        """
+        if not showall:
+            return self.api_root('projects')
+        else:
+            payload = 'showall'
+            return self.api_root('projects', parameters=payload)
+
+    def get_my_project(self, name=None, id=None):
+        """
+        Gets a specific MERMAID project. Either name or id required.
+        See https://mermaid-api.readthedocs.io/en/latest/projects.html for more information on project resources.
+        :param name: (optional if 'id' provided) MERMAID project name.
+        :type name: str
+        :param id: (optional if 'name' provided) MERMAID project ID.
+        :type id: str
+        :return: project with associated data.
+        :rtype: dict, None
+        """
+        if id:
+            return self.api_projects(id=id)
```

Comment:
If you accept my rename suggestion, this line could be more clearly `return self.project_resource(id=id, resource=None)`